packages/iTwinUI-react/src/core/ComboBox/helpers.ts

```diff
@@ -25,6 +33,30 @@ export const comboBoxReducer = (
         focusedIndex: value ?? state.focusedIndex,
       };
     }
+    case 'multiselect': {
+      if (action === 'removed') {
+        console.log('removed', value);
+        return {
+          ...state,
+          focusedIndex: value ?? state.focusedIndex,
+          selectedIndex: (state.selectedIndex as number[]).filter(
+            (index) => index !== value,
+          ),
+        };
+      }
+      if (action === 'added') {
+        console.log('added', value);
+        console.log([...(state.selectedIndex as number[]), value]);
+        return {
+          ...state,
+          focusedIndex: value ?? state.focusedIndex,
+          selectedIndex:
+            value !== null && value !== undefined
+              ? [...(state.selectedIndex as number[]), value].sort()
```

Comment:
the default `sort` function (without an argument) has quite unexpected results. Are you sure this is what you want?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort